app/main.py

```diff
@@ -1,54 +1,74 @@
+import xml
+import xml.etree.ElementTree as eT
 
 
 class Book:
-    def __init__(self, title: str, content: str):
+    def __init__(self, title: str, content: str) -> None:
         self.title = title
         self.content = content
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+
+class DisplayBook(Book):
+    def __init__(self, book: Book) -> None:
+        super().__init__(title=book.title, content=book.content)
+
+    def display_console(self) -> None:
+        print(self.content)
+
+    def display_reverse(self) -> None:
+        print(self.content[::-1])
+
+
+class PrintBook(Book):
+    def __init__(self, book: Book) -> None:
+        super().__init__(title=book.title, content=book.content)
+
+    def print_console(self) -> None:
+        print(f"Printing the book: {self.title}...")
+        print(self.content)
+
+    def print_reverse(self) -> None:
+        print(f"Printing the book in reverse: {self.title}")
+        print(self.content[::-1])
+
+
+class SerializeBook(Book):
+    def __init__(self, book: Book) -> None:
+        super().__init__(title=book.title, content=book.content)
+
+    def serialize_json(self) -> json:
+        return json.dumps({
+            "title": self.title,
+            "content": self.content
+        })
+
+    def serialize_xml(self) -> xml:
+        root = eT.Element("book")
+        title = eT.SubElement(root, "title")
+        title.text = self.title
+        content = eT.SubElement(root, "content")
+        content.text = self.content
+        return eT.tostring(root, encoding="unicode")
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    display_book = DisplayBook(book)
+    print_book = PrintBook(book)
+    serializer_book = SerializeBook(book)
```

Comment:
Do not initialize all instances from the beginning, make it depending on a command in the loop